tenpy/networks/site.py

```diff
@@ -222,6 +231,18 @@ def add_op(self, name, op, need_JW=False):
         self.opnames.add(name)
         if need_JW:
             self.need_JW_string.add(name)
+        # keep track of h.c. operators
+        if hc is None:
+            if op.conj().transpose() == op:
+                hc = name
+            else:
+                for other in self.opnames:
+                    other_op = self.get_op(other)
+                    if other_op.conj().transpose() == op:
+                        hc = other
```

Comment:
For performance, I will add a `break` here, so we stop as soon as we've found the `hc`.

I am thinking if it would also be worth adding a check for whether `hc_ops` already contains `name`. This would potentially avoid half the work (because we are adding both `name` and `hc` every time we call `add_op()`

Reply:
I don't think performance is critical here, since the onsite dimensions are usually small and the site creation is only done once in the very beginning of model creation.
However, a `break` certainly makes sense!